### ISSUE TITLE
In ruby 3.2.2 version, Watir::RSpec::HtmlFormatter will delete current project files

### ISSUE THREAD

**Issue by simonwu-os** [watir/watir-rspec#18]
In ruby 3.2.2 version.
```ruby
RSpec.configure do |config|
  config.add_formatter(Watir::RSpec::HtmlFormatter)
end
```

I don't set ENV["WATIR_RESULTS_PATH"] as  before versions.
Running " bundle exec rspec " twice,  It will delete all files from current project.
Because in ruby 3.2.2 version, the default output_wrapper will response to method :path , it returns "\<STDOUT\>".

```ruby
 @output_path = File.expand_path(ENV["WATIR_RESULTS_PATH"] || (output.respond_to?(:path) ? output.path : "tmp/spec-results/index.html"))
FileUtils.rm_rf File.dirname(@output_path), :verbose => true if File.exist?(@output_path)
```

@output_path  is assigned "<CURRENT_DIRECTORY>/\<STDOUT\>".
Run rspec first time, it will output "\<STDOUT\>" file in the current directory.
Run rspec second time, it will remove all files in the project directory.

It's very bad to delete all files in the current directory.

**Comment by jarmo** [comment]
First of all - really sorry that you had to deal with this unfortunate situation. I hope that nothing critical was deleted.

It's really bad indeed when something like this happens and we need to get it fixed ASAP.

I have a little bit problem of understanding the situation described. Do you know what is the type of `output` passed to `#initialize` exactly? What is the `<STDOUT>` part in the path exactly (e.g. what file is created?)? Do you know if problem happens only with Ruby 3.2.2 or is also happening with Ruby 3.1.2 for example?

**Comment by jarmo** [comment]
I guess looking at your description, then this is the `output` and the `#path` method - https://ruby-doc.org/3.2.2/IO.html#method-i-path

Ruby 3.1.2 does not have `IO#path`. Pretty scary indeed.

**Comment by jarmo** [comment]
Managed to reproduce this problem on Ruby 3.2.2 and yanked watir-rspec 4.0.0 to prevent any further accidental damage. 

_Thankfully_ 4.0.0 is the only version working with Ruby 3.2.2 thus far and one of the lines failing is the one responsible for deletion https://github.com/watir/watir-rspec/commit/97f5d193d7f05c259696a025e1ba1e11e9cceded#diff-60f968e3effe8372501a079d4edaeb5c8101362bdfa85e2267a85a90ddf09154L19, but it doesn't work due to usage of `File#exists?`.

And this design decision by Ruby core team is just awesome:
```ruby
irb(main):001:0> $stdout.path
=> "<STDOUT>"
```

**Comment by jarmo** [comment]
Released 5.0.0 with the above problem fixed. Thank you for reporting and I'm really sorry again that this happened to you.

To be honest - I would not write code like this today where deletion happens using possible user provided path (either via `WATIR_RESULTS_PATH` env variable or `output.path` which can be provided using RSpec options). I checked that this code was added at 2012 in commit d2a65730a31189874a616b8b35cc181a557497db.

Sorry again and I hope that you will continue using `watir-rspec`.

**Comment by simonwu-os** [comment]
That's ok. This time I happened to push my code. so I didn't loss any useful code.
This surprised me at first. I thought it was a virus that deleted these files. I kept trying to run rspec many times and found that it was caused by watir-rspec. 
Finally added ENV["WATIR_RESULTS_PATH"] = "tmp/spec-results/index.html" into the code to avoid this problem.
But I think it's serious problem, other users will encounter similar problem if they upgrade to ruby3.2, it's very bad problem.
